fe/fe-core/src/main/java/org/apache/doris/analysis/CreateTableStmt.java

```diff
@@ -406,6 +406,12 @@ public void analyze(Analyzer analyzer) throws UserException {
                 throw new AnalysisException("Create olap table should contain distribution desc");
             }
             distributionDesc.analyze(columnSet);
+            List<String> distributionColNames = ((HashDistributionDesc)distributionDesc).getDistributionColumnNames();
+            if (partitionDesc != null) {
+                if (checkIsDuplicate(partitionDesc.partitionColNames, distributionColNames)) {
+                    throw new AnalysisException("Partition column and bucket column should not have duplicates");
```

Comment:
I don't think we should restrict users from this kind of use.
The problem you described in issue should be fixed by modifying the prune logic?